agent/hcp/telemetry/otlp_transform.go

```diff
@@ -0,0 +1,168 @@
+package telemetry
```

Comment:
I was hoping to re-use the implementation by the opentelemetry-go library for the conversion between the OTEL metrics sdk to the OTLP proto.

Turns out that [package](https://github.com/open-telemetry/opentelemetry-go/tree/exporters/otlp/otlpmetric/otlpmetrichttp/v0.37.0/exporters/otlp/otlpmetric/internal/transform) is an internal one. Here's how it's [used](https://github.com/open-telemetry/opentelemetry-go/blob/2e54fbb3fede5b54f316b3a08eab236febd854e0/exporters/otlp/otlpmetric/internal/exporter.go#L54).

I used the same implementation but I was able to reduce the code for our use-case to the current file.